dmutils/s3.py

```diff
@@ -34,13 +34,26 @@ def bucket_short_name(self):
 
         return match.group(1)
 
-    def save(self, path, file, acl='public-read', move_prefix=None):
+    def save(self, path, file, acl='public-read', move_prefix=None, timestamp=None):
+        """Save a file in an S3 bucket
+
+        canned ACL list: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
+
+        :param path:        location in S3 bucket at which to save the file
+        :param file:        file object to be saved in S3
+        :param acl:         S3 canned ACL
+        :param move_prefix: Prefix to give to existing file when moving it out of the way
+        :param timestamp:   Timestamp to set for this file rather than using utcnow
+
+        :return: S3 Key
+        """
         path = path.lstrip('/')
 
         self._move_existing(path, move_prefix)
 
         key = self.bucket.new_key(path)
         filesize = get_file_size_up_to_maximum(file)
+        key.set_metadata('timestamp', (timestamp or datetime.datetime.utcnow()).isoformat())
```

Comment:
It seems weird to set the `timestamp` metadata here but then look at `last_modified` for the sort later (line #105) - are they both the same thing in S3?